jobboard/views.py

```diff
@@ -1,5 +1,76 @@
 from django.shortcuts import render
+from .models import Job
+from datetime import date
+from student.models import Student
+from recruiter.models import Recruiter
+
+
+def get_content_if_user_is_student(request):
+    """
+    If the user is a student then the jobs that will automatically
+    be displayed will be jobs that correspond to that student's degree
+    """
+    # use get methods once they get merged
+    user_as_student = Student.objects.get(user_id=request.user.id)
+    jobs_by_student_major = Job.get_jobs_by_major(user_as_student.major)
+    content = {'jobs_to_display': jobs_by_student_major,
+               'user': request.user,
+               'user_is_student': True,
+               'user_is_recruiter': False,
+               'user_as_student': user_as_student}
+    return content
+
+
+def get_content_if_user_is_recruiter(request):
+    """
+    If the user is a recruiter then the jobs that will automatically
+    be displayed will be jobs that this recruiter has created
+    """
+    # use get methods once they get merged
+    user_as_recruiter = Recruiter.objects.get(user_id=request.user.id)
+    jobs_created_by_recruiter = Job.get_jobs_by_recruiter_id(user_as_recruiter)
+    content = {'jobs_to_display': jobs_created_by_recruiter,
+               'user': request.user,
+               'user_is_student': False,
+               'user_is_recruiter': True,
+               'user_as_recruiter': user_as_recruiter}
+    return content
+
+
+def get_date_six_months_ago():
```

Comment:
1. why do you need this function?
2. I think that your function won't work for some special end cases, i.e for months with 31 days that 6 months earlier don't have 31 days
3. look up `datetime` - it'll be easier to do that with it

Reply:
Yes I see now that some edge cases won't work. I couldn't find any built in functions for this calculation with datetime but I will find a solution.